presto-hive/src/test/java/com/facebook/presto/hive/parquet/AbstractTestParquetReader.java

```diff
@@ -938,24 +939,63 @@ public void testDecimalBackedByINT64()
         }
     }
 
+    private void testDecimal(int precision, int scale, Optional<MessageType> parquetSchema)
+            throws Exception
+    {
+        ContiguousSet<BigInteger> values = bigIntegersBetween(BigDecimal.valueOf(Math.pow(10, precision - 1)).toBigInteger(), BigDecimal.valueOf(Math.pow(10, precision)).toBigInteger());
+        ImmutableList.Builder<SqlDecimal> expectedValues = new ImmutableList.Builder<>();
+        ImmutableList.Builder<HiveDecimal> writeValues = new ImmutableList.Builder<>();
+        for (BigInteger value : limit(values, 1_000)) {
+            writeValues.add(HiveDecimal.create(value, scale));
+            expectedValues.add(new SqlDecimal(value, precision, scale));
+        }
+        tester.testRoundTrip(new JavaHiveDecimalObjectInspector(new DecimalTypeInfo(precision, scale)),
+                writeValues.build(),
+                expectedValues.build(),
+                createDecimalType(precision, scale),
+                parquetSchema);
+    }
+
     @Test
-    public void testDecimalBackedByFixedLenByteArray()
+    public void testShortDecimalBackedByFixedLenByteArray()
+            throws Exception
+    {
+        for (int precision = 1; precision <= MAX_SHORT_PRECISION; precision++) {
+            int scale = ThreadLocalRandom.current().nextInt(precision);
```

Comment:
I get very nervous seeing random in tests. If something fails it's not reproducible, Use fixed constant test data instead, even if the fixed values are initially randomly chosen.